tools/brl_checks.h

```diff
@@ -22,84 +25,101 @@
+ * @param expected expected output
+ * @param typeform (optional) the typeform for this translation. If not specified it
+ * defaults to NULL.
+ * @param mode (optional) the translation mode. If not specified it defaults to 0.
+ * @param expected_inputPos (optional) the expected input positions. If not specified
+ * it defaults to NULL.
+ * @param expected_outputPos (optional) the expected input positions. If not specified
+ * it defaults to NULL.
+ * @param cursorPos (optional) the cursor position for this translation. If not specified
+ * it defaults to -1.
+ * @param expected_cursorPos (optional) the expected cursor position after this
+ * translation. If not specified it defaults to -1.
+ * @param direction (optional) 0 for forward translation, 1 for backwards translation. If
+ * not specified it defaults to 0.
+ * @param diagnostics (optional) Print diagnostic output on failure if diagnostics is not
+ * 0. If not specified it defaults to 1.
+ * @return Return 0 if the translation is as expected and 1 otherwise.
+ *
+ * An example how to invoke this is shown below:
+ * ~~~~~~~~~~~~~~~~~~~~~~
+ * result = check(table, word, expected_translation,
+ *                .typeform = typeform,
+ *                .cursorPos = 5);
+ * ~~~~~~~~~~~~~~~~~~~~~~
+ */
+#define check(tables, input, expected, ...)                                      \
+	check_base(tables, input, expected, (optional_test_params){.typeform = NULL, \
+												.cursorPos = -1,                 \
+												.expected_cursorPos = -1,        \
+												.expected_inputPos = NULL,       \
+												.expected_outputPos = NULL,      \
+												.mode = 0,                       \
+												.direction = 0,                  \
+												.diagnostics = 1,                \
+												__VA_ARGS__ })
 
 int
-check_outpos(const char *tableList, const char *str, const int *expected_poslist);
-
-/* Check if the cursor position is where you expect it to be after
- * translating str. Return 0 if the translation is as expected and 1
- * otherwise. */
+check_base(const char *tableList, const char *input, const char *expected,
+		optional_test_params in);
+
+/** Check the cursor positions for a translation
+ *
+ * For a given input string iterate over all initial cursor positions
+ * and check if the returned cursor position equals the one in
+ * expected_pos at the same index.
+ *
+ * Note: This check always translates with compbrlAtCursor and does
+ * not check the translation. This would not make sense anyway as the
+ * translation changes depending on the initial cursor position. For
+ * that reason this function is no longer used in checkyaml.
+ *
+ * @return 0 if the cursor position for each initial position in the
+ * input string equals the one in expected_pos at the same index and 1
+ * otherwise.
+ * @deprecated use the check() function instead
+ */
 int
 check_cursor_pos(const char *tableList, const char *str, const int *expected_pos);
```

Comment:
Is this function still used somewhere?

Reply:
Yes in the c-based tests that test the cursor pos. They rely on this weird behaviour that loops trough all possible cursor positions. Didn't feel like rewriting those.